uaa/src/test/java/org/cloudfoundry/identity/uaa/login/LoginMockMvcTests.java

```diff
@@ -694,6 +694,15 @@ void testCustomFavIcon() throws Exception {
                 .andExpect(content().string(allOf(containsString("<link href='data:image/png;base64,/sM4lL==' rel='shortcut icon' />"), not(containsString("square-logo.png")))));
     }
 
+    @Test
+    void testProductLogoOver100kChars() throws Exception {
+        String bigLogo = new String(new char[150000]).replace('\0', 'x');
+        setZoneFavIconAndProductLogo(webApplicationContext, identityZoneConfiguration, null, bigLogo);
+
+        mockMvc.perform(get("/login"))
+                .andExpect(content().string(allOf(containsString("<style>.header-image {background-image: url(data:image/png;base64," + bigLogo + ");}</style>"), not(containsString("product-logo.png")))));
```

Comment:
I guess this expect is not strictly required. All we need to validate is that the page loads correctly, don't we?
There already is a test that checks that the logo is present in the html just above.

Did you consider that?

Reply:
We also have to make sure this includes the biglogo, not the default fallback logo which happens if a concat returns a null.